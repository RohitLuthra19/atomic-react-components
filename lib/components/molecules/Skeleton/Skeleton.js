// @flow
/**
 *
 * Skeleton
 *
 */
import React from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import type { Node } from 'react';
import styles from './Skeleton.style';
import type { Props, LineProps, CircleProps } from './types';

const Circle = ({ className, width, margin }: CircleProps): Node => (
  <>
    <div
      className={classnames('skeletalCircle', className)}
      style={{ width, margin, height: width }}
    />
  </>
);

const Line = ({ className, height, width, margin }: LineProps): Node => (
  <>
    <div className={classnames('skeletalLine', className)} style={{ height, width, margin }} />
  </>
);

const Skeleton = ({ children, className }: Props): Node => (
  <div className={classnames('skeleton', className)}>{children}</div>
);

Skeleton.circle = Circle;
Skeleton.line = Line;

Skeleton.defaultProps = {};

Line.defaultProps = {
  height: '18px',
  width: '100%',
  margin: '0',
};

Circle.defaultProps = {
  width: '100px',
  margin: '20px',
};

export default styled(Skeleton)`
  ${styles};
`;

export { Skeleton as SkeletonVanilla };
