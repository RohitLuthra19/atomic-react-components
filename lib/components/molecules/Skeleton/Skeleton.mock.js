const defaultConfig = {
  className: 'test-line',
  lineHeight: '20px',
  lineWidth: '100%',
  lineGutter: '5px',
};
const circleConfig = {
  className: 'test-circle',
  circleGutter: '5px',
};

export { defaultConfig, circleConfig };
