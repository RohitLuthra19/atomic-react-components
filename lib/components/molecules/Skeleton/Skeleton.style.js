import { css } from 'styled-components';

export default css`
  @keyframes pulse {
    0% {
      background-position: 0% 0%;
    }
    100% {
      background-position: -135% 0%;
    }
  }
  .skeleton {
    div {
      display: inline-block;
      background: linear-gradient(-90deg, #ededed 0%, #e1e1e1 50%, #ededed 100%);
      background-size: 400% 400%;
      animation: pulse 1.2s ease-in-out infinite;
      margin: ${props => (props.gutter ? props.gutter : '0')};
      .skeletalLine {
        border-radius: 5px;
        height: ${props => (props.height ? props.height : '18px')};
        width: ${props => (props.width ? props.width : '100%')};
      }
      .skeletalCircle {
        border-radius: ${props => (props.radius ? props.radius : '50%')};
        height: ${props => (props.height ? props.height : '100px')};
        width: ${props => (props.height ? props.height : '100px')};
      }
    }
  }
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;
