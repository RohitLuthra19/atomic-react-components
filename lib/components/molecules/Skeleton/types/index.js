// @flow
import type { Node } from 'react';

export type Props = {
  children: Node,
  className?: string,
};

export type ShapesProps = {
  className?: string,
  radius?: number,
  height?: number,
  width?: number,
  gutter?: number,
  shapeType?: string,
};
