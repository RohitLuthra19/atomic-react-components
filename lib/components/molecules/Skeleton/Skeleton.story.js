import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import {
  defaultConfig,
  circleConfig,
  accordionConfig,
  tabsConfig,
  tabsContentConfig,
} from './Skeleton.mock';

// Import Styled Component to showcase variations
import Skeleton from '.';

storiesOf('Molecules/Skeleton', module)
  .addParameters({ jest: ['Skeleton', 'SkeletonVanilla'] })
  .add('Default Skeleton', () => (
    <Skeleton {...defaultConfig}>
      <Skeleton.bar {...defaultConfig}></Skeleton.bar>
      <Skeleton.bar {...defaultConfig}></Skeleton.bar>
      <Skeleton.circle {...defaultConfig}></Skeleton.circle>
      <Skeleton.circle {...circleConfig}></Skeleton.circle>
      <Skeleton.bar {...defaultConfig}></Skeleton.bar>
    </Skeleton>
  ))
  .add('Accordion Skeleton', () => (
    <Skeleton>
      <Skeleton.bar {...defaultConfig}></Skeleton.bar>
      <Skeleton.bar {...accordionConfig}></Skeleton.bar>
      <Skeleton.bar {...defaultConfig}></Skeleton.bar>
      <Skeleton.bar {...accordionConfig}></Skeleton.bar>
    </Skeleton>
  ))
  .add('Tabs Skeleton', () => (
    <Skeleton>
      <Skeleton.bar {...tabsConfig}></Skeleton.bar>
      <Skeleton.bar {...tabsConfig}></Skeleton.bar>
      <Skeleton.bar {...tabsContentConfig}></Skeleton.bar>
    </Skeleton>
  ));
