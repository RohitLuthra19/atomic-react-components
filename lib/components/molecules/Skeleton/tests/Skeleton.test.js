import React from 'react';
import { mount } from 'enzyme';

import Skeleton from '../index';

describe('<Skeleton />', () => {
  let SkeletonComponent = '';
  beforeEach(() => {
    SkeletonComponent = mount(
      <Skeleton>
        <Skeleton.line />
        <Skeleton.circle />
      </Skeleton>
    );
  });

  test('should render the <Skeleton> /', () => {
    expect(SkeletonComponent.exists()).toBe(true);
  });

  test('should render <div/>', () => {
    expect(SkeletonComponent.find('div')).toBeTruthy();
  });

  test('should render a skeleton line', () => {
    expect(SkeletonComponent.find('.skeletalLine')).toBeTruthy();
  });

  test('should render a skeleton circle', () => {
    expect(SkeletonComponent.find('.skeletalCircle')).toBeTruthy();
  });

  test('should render correctly', () => {
    expect(SkeletonComponent).toMatchSnapshot();
  });
});
