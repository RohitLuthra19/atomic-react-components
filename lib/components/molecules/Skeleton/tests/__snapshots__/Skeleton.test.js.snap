// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> should render correctly 1`] = `
<Skeleton>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "circle": [Function],
        "componentStyle": ComponentStyle {
          "componentId": "Skeleton-sc-1oak7zh-0",
          "isStatic": false,
          "lastClassName": "kAdMpY",
          "rules": Array [
            "@keyframes pulse{0%{background-position:0% 0%;}100%{background-position:-135% 0%;}}div{display:inline-block;background:linear-gradient(-90deg,#ededed 0%,#e1e1e1 50%,#ededed 100%);background-size:400% 400%;animation:pulse 1.2s ease-in-out infinite;&.skeletalLine{border-radius:5px;height:",
            [Function],
            ";width:",
            [Function],
            ";margin:",
            [Function],
            ";}&.skeletalCircle{border-radius:",
            [Function],
            ";height:",
            [Function],
            ";width:",
            [Function],
            ";margin:",
            [Function],
            ";}}",
            [Function],
            ";",
            ";",
          ],
        },
        "displayName": "Skeleton",
        "foldedComponentIds": Array [],
        "line": [Function],
        "render": [Function],
        "styledComponentId": "Skeleton-sc-1oak7zh-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <Skeleton
      className="Skeleton-sc-1oak7zh-0 kAdMpY"
    >
      <div
        className="skeleton Skeleton-sc-1oak7zh-0 kAdMpY"
      >
        <Line
          lineGutter="5px"
          lineHeight="20px"
          lineWidth="100%"
        >
          <div
            className="skeletalLine"
            skeletalLineGutter="5px"
            skeletalLineHeight="20px"
            skeletalLineWidth="100%"
          />
        </Line>
        <Circle
          circleGutter="5px"
          circleRadius="30px"
          circleWidth="100px"
        >
          <div
            className="skeletalCircle"
            skeletalCircleGutter="5px"
            skeletalCircleRadius="30px"
            skeletalCircleWidth="100px"
          />
        </Circle>
      </div>
    </Skeleton>
  </StyledComponent>
</Skeleton>
`;
